### DISCUSSION TITLE
Random slideshow and the ability to randomly display all resources

### DISCUSSION THREAD

**Discussion by joe128** [RouHim/this-week-in-past#152] [Ideas]
### Problem
If the config `RANDOM_SLIDESHOW` is enabled, a random image is fetched for each `slideshowTick`. So there is a very high chance that an image will be displayed twice before all resources have been displayed.  
This behaviour isn't suitable if you want to display a slideshow over a small number of images.  
(I have mentioned this issue and my usecase under https://github.com/RouHim/this-week-in-past/issues/138#issuecomment-2334762370)

### Solution
Perhaps we could introduce a config that allows a kind of "shuffle and display all resources" (SHUFFLE_ALL)?
_Another possibilities_
- refactor the existing config `RANDOM_SLIDESHOW`
- introduce a config `SHOW_ALL` (with optional `RANDOM_SLIDESHOW`)

I have already implemented a POC and tested it with a small set of images.

### Minimal config example to reproduce:
```
rm -rf testresources \
  && mkdir testresources \
  && rm -rf devdatafolder \
  && mkdir devdatafolder \
  && wget -O testresources/pic1.jpg https://picsum.photos/500/500 \
  && wget -O testresources/pic2.jpg https://picsum.photos/500/500 \
  && RESOURCE_PATHS=testresources DATA_FOLDER=devdatafolder RANDOM_SLIDESHOW=true SLIDESHOW_INTERVAL=3 ./this-week-in-past
```

**Comment by RouHim** [comment]
Sounds like a good idea, I've created an issue for the idea.

**Comment by joe128** [comment]
I think there are two ways to implement the feature "display all resources" and fix the issue "some images are displayed multiple times":

1. Introduce config SHUFFLE_ALL and leave RANDOM_SLIDESHOW untouched
2. Introduce config SHOW_ALL and optionally combine it with RANDOM_SLIDESHOW

| Configs |behaviour|further|option|
|--- | --- | --- | --- |
|SHUFFLE_ALL=true|  - backend serves all available image-ids in random order <br>  -  browser iterates over images | optional shuffle collection in browser after all images were shown | 1|
|RANDOM_SLIDESHOW=true|  unchanged | |1,2|
|SHOW_ALL=true + RANDOM_SLIDESHOW=false| - backend serves all available image-ids order by taken <br>  -  ui iterates over images | optional shuffle collection in browser after all images were shown|2|
|SHOW_ALL=true + RANDOM_SLIDESHOW=true| - backend serves all available image-ids order by taken <br>  - shuffle collection in browser <br> - browser iterates over images | optional shuffle collection in browser after all images were shown|2|


Both options should be implemented regardless of the “existence of images for the week”.

**Comment by hajo62** [comment]
I would like to extend this discussion or feature request to display all images matching the +/- 3 days before showing images already displayed. The work around discussed in [#138](https://github.com/RouHim/this-week-in-past/issues/138#issuecomment-1978163849) seems not to solve this.
